internal/cmd/commands/targetscmd/funcs.go

```diff
@@ -676,13 +676,20 @@ func printItemTable(result api.GenericResult) string {
 	maxLength := base.MaxAttributesLength(nonAttributeMap, item.Attributes, keySubstMap)
 
 	var hostSourceMaps []map[string]interface{}
+	// used as a key to ensure there are no repeat printed values among HostSources and HostSets
+	type setInfo struct {
+		id            interface{}
```

Comment:
Curious to why these are both `interface{}`.  Both `id` and `hostCatalogID` are strings

Reply:
I did it this way because in line 687 `m` is defined as `map[string]interface{}`

Reply:
Aha, so the CLI uses https://github.com/hashicorp/boundary/blob/main/internal/cmd/base/format.go#L119 to allow uniform rendering of maps.  Notice the input to that function is a `map[string]interface{}` here the use of `interface{}` is needed because the `WrapMap` function needs to deal embedded maps.  When possible its best to be specific about the type

Reply:
I see. Seems like I need to get better at looking through the code to figure out why things are written a certain way rather than just copying a format across changes